File: src/main/java/com/enderzombi102/cmt/zone/ZoneComponent.java
package com.enderzombi102.cmt.zone;

import dev.onyxstudios.cca.api.v3.component.ComponentV3;
import dev.onyxstudios.cca.api.v3.component.tick.ServerTickingComponent;
import net.minecraft.entity.Entity;
import net.minecraft.entity.player.PlayerEntity;
import org.jetbrains.annotations.Nullable;

import java.util.List;

public interface ZoneComponent extends ComponentV3, ServerTickingComponent {

	void addZone(AbstractZone<? extends Entity> zone);
	String removeZone(PlayerEntity entity);
	void removeZone(String id);
	boolean containsZone(String id);
	boolean entityInZone(List<Entity> entities, String id);
	@Nullable AbstractZone<? extends Entity> getZone(String id);
	List< AbstractZone<? extends Entity> > getZones();
	boolean isIdValid(String id);
	String getRandomId();
}

File: src/main/kotlin/com/enderzombi102/cmt/zone/ZoneComponent.kt
package com.enderzombi102.cmt.zone

import dev.onyxstudios.cca.api.v3.component.ComponentV3
import dev.onyxstudios.cca.api.v3.component.tick.ServerTickingComponent
import net.minecraft.entity.Entity
import net.minecraft.entity.player.PlayerEntity

interface ZoneComponent : ComponentV3, ServerTickingComponent {
	fun addZone(zone: AbstractZone<out Entity>)
	fun removeZone(entity: PlayerEntity): String?
	fun removeZone(id: String)
	fun containsZone(id: String): Boolean
	fun entityInZone(entities: List<Entity>, id: String): Boolean
	fun getZone(id: String): AbstractZone<out Entity>?
	fun getZones(): List<AbstractZone<out Entity>>
	fun isIdValid(id: String): Boolean
	val randomId: String
}
